zipline/pipeline/factors/basic.py

```diff
@@ -489,6 +491,38 @@ def compute(self, today, assets, out, returns, annualization_factor):
         out[:] = nanstd(returns, axis=0) * (annualization_factor ** .5)
 
 
+class PeerCount(SingleInputMixin, CustomFactor):
+    """
+    Peer Count of distinct categories in a given classifier.  This factor
+    is returned by the classifier instance method peer_count()
+
+    **Default Inputs:** None
+
+    **Default Window Length:** 1
+    """
+    window_length = 1
+    missing_value = -1
+    dtype = int64_dtype
+
+    def _validate(self):
+        super(PeerCount, self)._validate()
+        if self.window_length != 1:
+            raise ValueError(
+                "'PeerCount' expected a window length of 1, but was given"
+                "{window_length}.".format(window_length=self.window_length)
+            )
+
+    def compute(self, today, assets, out, arrays):
+        # Convert arrays to group label int arrays
+        group_labels, _ = self.inputs[0]._to_integral(arrays[0])
```

Comment:
Not your doing, but I'm surprised to see `_to_integral` as a private method. I suppose we don't expect users to ever call it on classifiers in an actual algorithm.

Reply:
Yeah, I've only seen it used two other places and both were in ._compute() functions.  Not sure if it is bad practice for me to call a private method in CustomFactor's .compute()?

Reply:
I could also hand compute _to_integral() pretty easily in the compute() function

Reply:
I think given its uses elsewhere, what you have here is fine.